WordPress/openverse
# Upgrade eslint-plugin-playwright to get new checks; fix small checkbox bug

## Description

These are changes I made while investigating https://github.com/WordPress/openverse/issues/4087. The issue turned out not to be actually an issue, but these new tests are useful, and I found a different bug anyway, which is fixed in this PR. I also upgraded us to the latest playwright ESLint plugin, and I will need to fix the many new warnings this has produced. I did the upgrade originally hoping that https://github.com/playwright-community/eslint-plugin-playwright/blob/main/docs/rules/prefer-web-first-assertions.md would catch `toHaveAttribute("checked", "checked")`, but of course, that's a different assertion than `toBeChecked()`. Very easy to get wrong! I'm sure I learned this at some point, but it's a trivial thing to forget.

There was also an issue with the TypeScript version we use in the application being outside the supported range of the older TS ESLint plugin we used. I've updated the TS ESLint plugin, which also required updating ESLint, which in turn required updating the other ESLint plugins. In the end I decided to just update all the ESLint stuff using `pnpm -r up 'eslint*'`.

## Testing Instructions

Run storybook using `just p frontend storybook` and visit the checkbox story: http://localhost:54000/iframe.html?id=components-vcheckbox--default-story

Confirm you can still click to toggle the checkbox on and off.

Run the new storybook tests (or look at them in CI) and confirm the assertions pass. If you run them locally, you can remove `pointer-events-none` from the checked icon to see the bug I fixed.

Run `pnpm run eslint` and see that there are no errors. CI will also confirm this.

## Checklist


- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
